tests/instruments/test_all_instruments.py

```diff
@@ -54,7 +54,7 @@
 init = ["ThorlabsPM100USB", "Keithley2700", "TC038", "Agilent34450A",
         "AWG401x_AWG", "AWG401x_AFG", "VARX", "HP8116A"]
 # Instruments which require more input arguments
-init.extend(["Instrument", "ATSBase"])
+init.extend(["Instrument", "ATSBase", "SPDBase"])
```

Comment:
What happens if you don't do that? From what I can see, `SPDBase` does _not_ need communication in its init?

Reply:
This is for the same reason as `ATSBase`. Neither of these two base classes provide an instrument name, which consistently causes the __init__ of `Instrument` to fail. This is not needed for the child classes because the name is defined in there. The other option I tried was to set a base name (something like Siglent SPDxxxxX instrument Base Class) but that caused further errors in the child classes with multiple `name` definitions

Reply:
@bilderbuchi the comment above the list states the reason for the exception: these instruments require input arguments (in this case `name`).

I know, that the skip message is not appropriate, but I did not add a special check just for `Instrument` class.

Reply:
My only excuse is that it must have been too late to notice that comment :-P

What if you add `kwargs.setdefault('name', 'Siglent SPDxxxxX instrument Base Class')` in `SPDBase`'s init, just before you call `super()__init__`? This should enable using it in tests, but still result in the correct names for the child classes?

Reply:
This works